src/js/structs/List.js

```diff
@@ -2,13 +2,33 @@ import Item from './Item';
 import StringUtil from '../utils/StringUtil';
 import Util from '../utils/Util';
 
+/**
+ * Cast an item to the List's type, if specified.
+ *
+ * Note that cast must be bound to the List's context in order to access
+ * its `type` property.
+ *
+ * @param {Object} item
+ * @access private
+ * @memberOf List
+ * @return {Object} item, cast if list is typed.
+ */
+function cast(item) {
+  let Type = this.constructor.type;
+  if (Type != null && !(item instanceof Type)) {
```

Comment:
wait, maybe i'm missing something, but won't this force all items of the `List` to be `List`s?